cdap-watchdog-api/src/main/java/io/cdap/cdap/api/metrics/MetricValue.java

```diff
@@ -24,12 +24,64 @@
   private MetricType type;
   private long value;
 
+  // following fields to support distribution aka histogram/event metrics
+  /*
+  Based on https://cloud.google.com/monitoring/api/ref_v3/rest/v3/TypedValue#Exponential
+   */
+  public static final int GROWTH_FACTOR = 2;
+
+  // Total buckets = NUM_FINITE_BUCKETS +2 which will be 64. A long number
+  // maybe used as a mask to indicate which buckets have non zero bucket counts
+  public static final int NUM_FINITE_BUCKETS = 62;
+
+  /**
+   * Based on https://cloud.google.com/monitoring/api/ref_v3/rest/v3/TypedValue#Distribution
+   *
+   * bucketCounts stores counts of buckets with non zero values.
+   */
+  private long[] bucketCounts;
+
+  // Most of the buckets have zero counts. For efficiency, bucketCounts only
+  // stores non zero bucket count values. The bucketMask stores which
+  // buckets have non-zero bucketCounts.
+  private long bucketMask = 0;
+
+  /**
+   * The number of values in the population. Must be non-negative.
+   * This value must equal the sum of the values in bucketCounts
+  */
+  private long count;
+
+  /**
+   * The arithmetic mean of the values in the population. If count is zero then this field must be zero.
+   */
+  private double mean;
+
+  /**
+   * The sum of squared deviations from the mean of the values in the population.
+   */
+  private double sumOfSquaredDeviation;
```

Comment:
I think you can name it Variance. How are you going to aggregate it?

Reply:
Streamz distribution has an online algo (from Knuth apparently). I didn't go into the details yet. Here is the link https://source.corp.google.com/piper///depot/google3/java/com/google/monitoring/streamz/Distribution.java;l=385?q=lang:java%20distrib&sq=package:piper%20file:%2F%2Fdepot%2Fgoogle3%20-file:google3%2Fexperimental

Reply:
This is for the original computation where you are adding the values. But down the road you would need to aggregate two MetricValue objects (e.g. for two successive seconds). How would you do it? Also, we did not discuss how are you going to store this in the MetricsStore.

Reply:
And yes, I refreshed my math memory, variance is a different thing (it's sumOfSquaredDeviation/(count-1))

Reply:
Let's just skip in this PR. We can add it later after we got the answers

Reply:
Removed.